protocol/x/accountplus/ante/ante.go

```diff
@@ -0,0 +1,298 @@
+package ante
+
+import (
+	"bytes"
+	"strconv"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	authante "github.com/cosmos/cosmos-sdk/x/auth/ante"
+
+	txsigning "cosmossdk.io/x/tx/signing"
+
+	"github.com/dydxprotocol/v4-chain/protocol/lib/metrics"
+	"github.com/dydxprotocol/v4-chain/protocol/x/accountplus/authenticator"
+	"github.com/dydxprotocol/v4-chain/protocol/x/accountplus/keeper"
+	"github.com/dydxprotocol/v4-chain/protocol/x/accountplus/types"
+)
+
+// AuthenticatorDecorator is responsible for processing authentication logic
+// before transaction execution.
+type AuthenticatorDecorator struct {
+	accountPlusKeeper *keeper.Keeper
+	accountKeeper     authante.AccountKeeper
+	sigModeHandler    *txsigning.HandlerMap
+	cdc               codec.Codec
+}
+
+// NewAuthenticatorDecorator creates a new instance of AuthenticatorDecorator with the provided parameters.
+func NewAuthenticatorDecorator(
+	cdc codec.Codec,
+	accountPlusKeeper *keeper.Keeper,
+	accountKeeper authante.AccountKeeper,
+	sigModeHandler *txsigning.HandlerMap,
+) AuthenticatorDecorator {
+	return AuthenticatorDecorator{
+		accountPlusKeeper: accountPlusKeeper,
+		accountKeeper:     accountKeeper,
+		sigModeHandler:    sigModeHandler,
+		cdc:               cdc,
+	}
+}
+
+// AnteHandle is the authenticator ante handler responsible for processing authentication
+// logic before transaction execution.
+func (ad AuthenticatorDecorator) AnteHandle(
+	ctx sdk.Context,
+	tx sdk.Tx,
+	simulate bool,
+	next sdk.AnteHandler,
+) (newCtx sdk.Context, err error) {
+	defer metrics.ModuleMeasureSince(
+		types.ModuleName,
+		metrics.AuthenticatorDecoratorAnteHandleLatency,
+		time.Now(),
+	)
+
+	// Make sure smart account is active.
+	if active := ad.accountPlusKeeper.GetIsSmartAccountActive(ctx); !active {
+		return ctx, types.ErrSmartAccountNotActive
+	}
```

Comment:
this is internalized. see the other [comment](https://github.com/dydxprotocol/v4-chain/pull/2325/files#r1772014493)

Reply:
Link doesn't work for me - which comment were you referring to/

Reply:
oh the big comment https://github.com/dydxprotocol/v4-chain/pull/2325#discussion_r1772014493